sklearn/linear_model/coordinate_descent.py

```diff
@@ -977,6 +977,10 @@ def _path_residuals(X, y, train, test, path, path_params, alphas=None,
     y_train = y[train]
     X_test = X[test]
     y_test = y[test]
+    if not sparse.issparse(X):
+        # fancy indexing of read-only memmap creates a read-only copy.
```

Comment:
Maybe add link reference to numpy#14132

Reply:
Just to be safe, I would only do this if we are sure that `X_train.base is not X`. Granted fancy indexing makes copies, but I would rather enforce it in code than in a comment,
```py
for array, array_input in ((X_train, X), (y_train, y), (X_test, X), (y_test, y)):
    if array.base is not array_input and not array.flags['WRITABLE']:
        # fancy indexing should create a writable copy but it doesn't for read-only
        # memmaps (cf. numpy#14132).
        array.setflags(write=True)
```